new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java

```diff
@@ -0,0 +1,854 @@
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Checking managed server service {0} is created in namespace {1}",
+          managedServerPodNamePrefix + i, introDomainNamespace);
+      checkServiceExists(managedServerPodNamePrefix + i, introDomainNamespace);
+    }
+
+    // get the pod creation time stamps
+    LinkedHashMap<String, DateTime> pods = new LinkedHashMap<>();
+    // get the creation time of the admin server pod before patching
+    DateTime adminPodCreationTime = getPodCreationTime(introDomainNamespace, adminServerPodName);
+    pods.put(adminServerPodName, adminPodCreationTime);
+    // get the creation time of the managed server pods before patching
+    for (int i = 1; i <= replicaCount; i++) {
+      pods.put(managedServerPodNamePrefix + i,
+          getPodCreationTime(introDomainNamespace, managedServerPodNamePrefix + i));
+    }
+
+    logger.info("change the cluster size and verify the introspector runs and updates the domain status");
+    // create a temporary WebLogic WLST property file
+    File wlstPropertiesFile = assertDoesNotThrow(() -> File.createTempFile("wlst", "properties"),
+        "Creating WLST properties file failed");
+    Properties p1 = new Properties();
+    p1.setProperty("admin_host", K8S_NODEPORT_HOST);
+    p1.setProperty("admin_port", Integer.toString(t3ChannelPort));
+    p1.setProperty("admin_username", ADMIN_USERNAME_DEFAULT);
+    p1.setProperty("admin_password", ADMIN_PASSWORD_DEFAULT);
+    p1.setProperty("cluster_name", clusterName);
+    p1.setProperty("max_cluster_size", Integer.toString(3));
+    p1.setProperty("test_name", "change_server_count");
+    assertDoesNotThrow(() -> p1.store(new FileOutputStream(wlstPropertiesFile), "wlst properties file"),
+        "Failed to write the WLST properties to file");
+
+    // change the server count of the cluster by running online WLST
+    Path configScript = Paths.get(RESOURCE_DIR, "python-scripts", "introspect_version_script.py");
+    executeWLSTScript(configScript, wlstPropertiesFile.toPath(), introDomainNamespace);
+
+    // patch the domain to increase the replicas of the cluster and add introspectVersion field
+    String patchStr =
+          "["
+            + "{\"op\": \"replace\", \"path\": \"/spec/clusters/0/replicas\", \"value\": 3},"
+            + "{\"op\": \"add\", \"path\": \"/spec/introspectVersion\", \"value\": \"2\"}"
+        + "]";
+
+    logger.info("Updating replicas in cluster {0} using patch string: {1}", clusterName, patchStr);
+    V1Patch patch = new V1Patch(patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, introDomainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+
+    //verify the introspector pod is created and runs
+    logger.info("Verifying introspector pod is created, runs and deleted");
+    String introspectPodName = domainUid + "-" + "introspect-domain-job";
+    checkPodExists(introspectPodName, domainUid, introDomainNamespace);
```

Comment:
I think this will be fine, but I'm wondering if we can eventually do better... What I mean is that this could fail in two different ways: 1) unlikely -- but the introspector could run very fast and the test would miss that the pod ever existed, and 2) the introspector could run but fail quickly and similarly be missed. Let's see if this test starts to fail intermittently.

Reply:
Yes I agree when I was manually testing it I missed the introspector lifecycle and noticed that within 30 seconds it finished running.
The withStandardRetryPolicy has a poll delay of 2 seconds so hopefully it shouldn't miss it. But lets see if fails intermittently.